command.go

```diff
@@ -613,13 +613,37 @@ Loop:
 
 // argsMinusFirstX removes only the first x from args.  Otherwise, commands that look like
 // openshift admin policy add-role-to-user admin my-user, lose the admin argument (arg[4]).
-func argsMinusFirstX(args []string, x string) []string {
-	for i, y := range args {
-		if x == y {
-			ret := []string{}
-			ret = append(ret, args[:i]...)
-			ret = append(ret, args[i+1:]...)
-			return ret
+// Special care needs to be taken not to remove a flag value.
+func argsMinusFirstX(args []string, x string, c *Command) []string {
+	if len(args) == 0 {
+		return args
+	}
+	c.mergePersistentFlags()
+	flags := c.Flags()
+
+Loop:
+	for pos := 0; pos < len(args); pos++ {
```

Comment:
Is there a particular reason to use `for pos := 0; pos < len(args); pos++` instead of `for i, y := range args`?

Reply:
Sorry, I didn't see you comment until now.

Yes, there is a reason:

Inside the loop, it can advance the position (`pos++` on line 636) in order to handle when there is a flag with an arg and an equal sign is not used (ie `--foo bar`).  In this case it needs to skip over the next token because it is the value of the flag, not an arg.